### PR TITLE
NAS-129639 / 25.04 / Remove unused API endpoints

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

**Comment by themylogin** [comment]
http://jenkins.eng.ixsystems.net:8080/job/master/job/custom/535/

File: src/middlewared/middlewared/plugins/crypto_/cert_info.py [hunk 1]
```diff
@@ -47,17 +47,6 @@ async def ec_curve_choices(self):
         """
         return {k: k for k in EC_CURVES}
 
-    @accepts()
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/crypto_/cert_info.py:50]: Are we certain UI is not using these ?

File: src/middlewared/middlewared/plugins/crypto_/cert_profiles.py [hunk 2]
```diff
@@ -105,15 +105,3 @@ async def profiles(self):
         configurations which can be used for creating certificates.
         """
         return CERTIFICATE_PROFILES
-
-    @accepts(roles=['CERTIFICATE_READ'])
-    @returns(Dict(
-        *[Dict(profile, additional_attrs=True) for profile in CSR_PROFILES],
-        example=CSR_PROFILES,
-    ))
-    async def certificate_signing_requests_profiles(self):
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/crypto_/cert_profiles.py:114]: UI should be using this for CSR profiles, seems like they are using the wrong endpoint atm

File: src/middlewared/middlewared/plugins/device_/device_info.py [hunk 3]
```diff
@@ -291,19 +291,3 @@ def get_gpus(self):
...
         return gpus
-
-    @accepts()
-    @returns(Dict(
-        'gpu_pci_id_choices',
-        additional_attrs=True,
-        description='Returns PCI id(s) of GPU(s) located in the system',
-        example={'Red Hat, Inc. QXL paravirtual graphic card': '0000:00:02.0'}
-    ))
-    async def gpu_pci_ids_choices(self):
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/device_/device_info.py:302]: UI should be using this as well iirc, @Qubad786 can you confirm please ?

File: src/middlewared/middlewared/plugins/system_advanced/gpu.py [hunk 4]
```diff
@@ -9,22 +8,6 @@ class Config:
         namespace = 'system.advanced'
         cli_namespace = 'system.advanced'
 
-    @accepts(roles=['SYSTEM_ADVANCED_READ'])
-    @returns(Dict(additional_attrs=True))
-    def get_gpu_pci_choices(self):
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/system_advanced/gpu.py:14]: This is being used by UI as well iirc, @Qubad786 can you confirm ?

**Review by sonicaj — COMMENTED** [review]
Had a quick pass at crypto/kmip/vm plugins

File: src/middlewared/middlewared/plugins/crypto_/cert_profiles.py (hunk 2 above)
Comment (themylogin) [reply src/middlewared/middlewared/plugins/crypto_/cert_profiles.py:114]: @undsoft it is strange that the UI does not use this

File: src/middlewared/middlewared/plugins/device_/device_info.py (hunk 3 above)
Comment (Qubad786) [reply src/middlewared/middlewared/plugins/device_/device_info.py:302]: This should be safe to remove

File: src/middlewared/middlewared/plugins/system_advanced/gpu.py (hunk 4 above)
Comment (Qubad786) [reply src/middlewared/middlewared/plugins/system_advanced/gpu.py:14]: It seems UI is using `device.get_info` right now but it should be using this endpoint instead and i think we already have a ticket for UI to use this endpoint

File: src/middlewared/middlewared/plugins/crypto_/cert_info.py (hunk 1 above)
Comment (undsoft) [reply src/middlewared/middlewared/plugins/crypto_/cert_info.py:50]: Yes, this is hardcoded. We can call an endpoint if we feel like that would be beneficial.

File: src/middlewared/middlewared/plugins/crypto_/cert_profiles.py (hunk 2 above)
Comment (undsoft) [reply src/middlewared/middlewared/plugins/crypto_/cert_profiles.py:114]: We call `webui.crypto.certificate_profiles`.

File: src/middlewared/middlewared/plugins/system_advanced/gpu.py (hunk 4 above)
Comment (undsoft) [reply src/middlewared/middlewared/plugins/system_advanced/gpu.py:14]: Ticket https://ixsystems.atlassian.net/browse/NAS-130983

**Review by creatorcary — APPROVED** [review]

File: src/middlewared/middlewared/plugins/crypto_/cert_profiles.py (hunk 2 above)
Comment (sonicaj) [reply src/middlewared/middlewared/plugins/crypto_/cert_profiles.py:114]: That is not accurate and UI should be using this endpoint instead, i'll make a ticket

**Review by sonicaj — COMMENTED** [review]
Let's please remove the endpoints which UI is going to consume, apart from that LGTM

File: src/middlewared/middlewared/plugins/crypto_/cert_profiles.py (hunk 2 above)
Comment (undsoft) [reply src/middlewared/middlewared/plugins/crypto_/cert_profiles.py:114]: Deja vu :) https://ixsystems.atlassian.net/browse/NAS-126028

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/crypto_/cert_profiles.py:114]: We missed specifying CSR profiles in that ticket, created a ticket for that https://ixsystems.atlassian.net/browse/NAS-131460

Comment (yocalebo) [reply src/middlewared/middlewared/plugins/crypto_/cert_profiles.py:114]: Please rename that function and also put it under webui namespace. webui.crypto.csr_profiles

**Comment by themylogin** [comment]
time 20:00

**Pull request merged**